tests/parametric/test_dynamic_configuration.py

```diff
@@ -801,3 +820,80 @@ def test_trace_sampling_rules_with_tags(self, test_agent, test_library):
         span = trace[0]
         assert "_dd.p.dm" in span["meta"]
         assert span["meta"]["_dd.p.dm"] == "-12"
+
+    @parametrize("library_env", [{**DEFAULT_ENVVARS}])
+    def test_remote_sampling_rules_telemetry(self, library_env, test_agent, test_library):
+        """RC sampling rules should be reported by telemetry"""
+        sampling_rule = [
+            {
+                "service": "svc*",
+                "resource": "*abc",
+                "name": "op-??",
+                "tags": [
+                    {"key": "tag-a", "value_glob": "ta-v*"},
+                    {"key": "tag-b", "value_glob": "tb-v?"},
+                    {"key": "tag-c", "value_glob": "tc-v"},
+                ],
+                "sample_rate": 0.5,
+                "provenance": "dynamic",
+            }
+        ]
+        sampling_rule_without_tags = [
+            {"service": "svc*", "resource": "*abc", "name": "op-??", "sample_rate": 0.5, "provenance": "dynamic",}
+        ]
+        sampling_rule_with_modified_tags = [
+            {
+                "service": "svc*",
+                "resource": "*abc",
+                "name": "op-??",
+                "tags": {"tag-a": "ta-v*", "tag-b": "tb-v?", "tag-c": "tc-v"},
+                "sample_rate": 0.5,
+                "provenance": "dynamic",
+            }
+        ]
+        telemetry_names = set(
+            [
+                "sampling_rules",
+                "trace.sampling_rules",
+                "DD_TRACE_SAMPLING_RULES",
+                "trace_sampling_rules",
+                "trace_sample_rules",
+            ]
+        )
+        event = set_and_wait_rc_telemetry(test_agent, config_overrides={"tracing_sampling_rules": sampling_rule,},)
+        configuration = event["payload"]["configuration"]
+        rules = next(filter(lambda x: x["name"] in telemetry_names, configuration))["value"]
+        if context.library.library == "nodejs":
+            assert json.loads(rules) == sampling_rule_without_tags
+        elif context.library.library == "python":
+            assert json.loads(rules) == sampling_rule_with_modified_tags
+        elif context.library.library == "golang":
+            assert json.loads(rules) == sampling_rule_with_modified_tags
+        elif context.library.library == "ruby":
+            assert rules == sampling_rule_with_modified_tags
+        else:
+            assert json.loads(rules) == sampling_rule
```

Comment:
That's against the system-tests philosophy. The goal of system-tests is to assert all tracers behave the same. I do not think we should accept different behavior per tracer for a new feature. 

In order words, you have to implement the specified behavior. If there is no specification about how the telemetry value for sampling rules should look like, then specify the behavior and force other libraries to implement the spec.

Reply:
It's more that we don't have a unified way of sending config values through telemetry. Some libraries send the original config value that the user inputs, others send the config value that is actually set on the tracer after some preliminary processing or validation. This becomes a problem with sampling rules since nodejs is ignoring tags so they are not included, libraries that support tags change the tags' format due to security considerations before setting them on the tracer. So this is a larger problem that we cannot unify across the languages right now. Should we disable the test for now?